pika/adapters/select_connection.py

```diff
@@ -27,20 +27,25 @@
 ERROR = 0x0008
 
 
-if pika.compat.PY2:
-    _SELECT_ERROR = select.error
-else:
-    # select.error was deprecated and replaced by OSError in python 3.3
-    _SELECT_ERROR = OSError
 
+_SELECT_ERRORS = (select.error, OSError, IOError)
 
-def _get_select_errno(error):
-    if pika.compat.PY2:
-        assert isinstance(error, select.error), repr(error)
-        return error.args[0]
-    else:
-        assert isinstance(error, OSError), repr(error)
-        return error.errno
+_SELECT_ERROR_CHECKERS = {
+    select.error : lambda e: e.args[0] == errno.EINTR,
+    OSError      : lambda e: e.errno == errno.EINTR,
+    IOError      : lambda e: e.errno == errno.EINTR,
+    }
```

Comment:
@shinji-s: Regarding
> And we are supposed to get no EINTR exceptions raised in Python 3.5 as the standard library performs retries. Some of the testes here should fail.

Let's add the appropriate `unittest.skip*` decorator to your test(s) to make sure that the one(s) that are not appropriate for Python 3.5 won't run on 3.5 and later.